### PR TITLE
Tests for Shared roster service and multi user chat service and refactor.

### PR DESCRIPTION
**Pull request by nandorholozsnyak**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -108,6 +109,13 @@
 				<version>${project.version}</version>
 			</dependency>
 
+			<dependency>
+				<groupId>org.glassfish.jersey.core</groupId>
+				<artifactId>jersey-common</artifactId>
+				<version>${jersey.common.version}</version>
```
Comment (kfnorbi) [inline pom.xml]: We probably should try this with Jackson

Comment (nandorholozsnyak) [reply pom.xml]: It was added because of Response class for the JUnit tests, I don't know why JAX RS API was not enough for it.

File: pom.xml [hunk 2]
```diff
@@ -108,6 +109,13 @@
 				<version>${project.version}</version>
 			</dependency>
 
+			<dependency>
+				<groupId>org.glassfish.jersey.core</groupId>
```
Comment (gabichelsea) [inline pom.xml]: I think we do not need to introduce a glassfish dependency

File: smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/ejabberd/SharedRosterServiceImpl.java [hunk 3]
```diff
@@ -127,8 +124,7 @@ public void deleteUserFromGroup(String user, String group) {
         .host(this.getXmppDomainPropertyValue())
         .group(group)
         .build();
-    return this.clientResponseProvider.sendPostRequest(SHARED_ROSTER_GROUP_INFO_COMMAND, informationRequest)
-        .readEntity(MAP_GENERIC_TYPE);
+    return (Map<String, String>) this.clientResponseProvider.sendPostRequest(SHARED_ROSTER_GROUP_INFO_COMMAND, informationRequest).getEntity();
```
Comment (gabichelsea) [inline smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/ejabberd/SharedRosterServiceImpl.java]: Why did you change it?

Comment (nandorholozsnyak) [reply smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/ejabberd/SharedRosterServiceImpl.java]: Test thrown exception for readEntity, "Exception in thread "main" java.lang.IllegalStateException: Method not supported on an outbound message. "

I found this kind of workaround for this problem so I had to replace it with getEntity.

**Comment by gabichelsea** [comment]
Please resolve the conflicts
